clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2805,6 +2749,13 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
 
     const auto *RecordDecl = Ty->getAsCXXRecordDecl();
     createSpecialMethodCall(RecordDecl, getInitMethodName(), BodyStmts);
+    CXXMethodDecl *FinalizeMethod =
+        getMethodByName(RecordDecl, FinalizeMethodName);
+    // A finalize-method is expected for stream classes.
```

Comment:
NIT: there is only one stream class in SYCL spec.
```suggestion
    // A finalize-method is expected for stream class.
```